util/app-config/src/lib.rs

```diff
@@ -217,17 +215,7 @@ fn is_daemon(subcommand_name: &str) -> bool {
 }
 
 fn consensus_from_spec(spec: &ChainSpec) -> Result<Consensus, ExitCode> {
-    spec.to_consensus().and_then(verify_genesis).map_err(|err| {
-        eprintln!("{:?}", err);
-        ExitCode::Config
-    })
-}
-
-fn verify_genesis(consensus: Consensus) -> Result<Consensus, Box<Error>> {
-    MerkleRootVerifier::new()
-        .verify(consensus.genesis_block())
-        .map_err(Box::new)?;
-    Ok(consensus)
+    spec.to_consensus().map_err(|_| ExitCode::Config)
```

Comment:
`spec/src/lib.rs` only ensures that genesis is matched with the specs, recorded in `specs/*.toml`.

`src/subcommand/run.rs` do the real block validation (Unable to do this inside `spec/src/lib.rs` cause circular dependency of `ckb_shared` and `ckb-chain-specs`).